tests/unit/src/test/scala/tests/WindowStateDidChangeSlowSuite.scala

```diff
@@ -0,0 +1,35 @@
+package tests
+
+object WindowStateDidChangeSlowSuite extends BaseSlowSuite("window-state-did-change") {
+  testAsync("compile-after-focus") {
+    for {
+      _ <- server.initialize(
+        """
+          |/metals.json
+          |{ "a": {} }
+          |/a/src/main/scala/a/A.scala
+          |package a
+          |object A {
+          |  val x = 42
+          |}
+        """.stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/A.scala")
+      _ = assertNoDiagnostics()
+      _ = server.windowStateDidChange(false)
+      didSave = server.didSave("a/src/main/scala/a/A.scala")(
+        _.replaceAllLiterally("val x", "x")
+      )
+      _ = assertNoDiagnostics()
```

Comment:
I can think about it if you want. Or even better think about it after delivering this functionality.

Reply:
Sleeping is fine. The test currently may not be stressing the pause functionality because the build doesn’t have brought time to publish diagnostics before the assertion

Reply:
Ok, I'll do in that way.